### PR TITLE
Rejig - Oleg-starter-pack

### PR DESCRIPTION
**Pull request by theartofnowt**
Ticket: https://webjobs.agepartnership.co.uk/desk/tickets/5152475/messages
Linked to: https://github.com/AgePartnership/web.agepartnership.co.uk/pull/2373

Preview checks:
https://agepartnership.atlassian.net/wiki/spaces/WTD/pages/1360101401/Widget+URLs

live: https://www.agepartnership.co.uk/equity-release/landing/201811-search-journey/stage-1/
docker: https://local.agepartnership.co.uk:4433/equity-release/landing/201811-search-journey/stage-1/

live: https://www.agepartnership.co.uk/equity-release/landing/201811-search-journey/diqf-eligibility/
docker : https://local.agepartnership.co.uk:4433/equity-release/landing/201811-search-journey/diqf-eligibility/

live: https://www.agepartnership.co.uk/equity-release/landing/201811-search-journey/diqf-ltv/
docker: https://local.agepartnership.co.uk:4433/equity-release/landing/201811-search-journey/diqf-ltv/

live: https://www.agepartnership.co.uk/equity-release/landing/201811-search-journey/diqf-youngest-homeowner/
docker: https://local.agepartnership.co.uk:4433/equity-release/landing/201811-search-journey/diqf-youngest-homeowner/

live: https://www.agepartnership.co.uk/equity-release/
docker: https://local.agepartnership.co.uk:4433/equity-release/

live: https://www.agepartnership.co.uk/pension-income/
docker: https://local.agepartnership.co.uk:4433/pension-income/

live: https://www.agepartnership.co.uk/styleguide/
docker: https://local.agepartnership.co.uk:4433/styleguide/

*Done*

- I've renamed _base.scss to _config.scss
- Created a _utilities.scss base file 
- Created a _variables.scss base file 
- Created a _config.scss base file 
- Put the _config/utilities/variables.thing.scss classes into a config/utilities/variables folder
- create a new mixins for the clearfix classes
- update readme
- sort indention on all files

Checks:

Best FE practice in terms of naming conventions, folder configurations
Malicious code

### CODE REVIEW

**Comment by theartofnowt** [comment]
where would you prefer the previews to be? in the agep repo pr or here?

**Comment by ntjnh** [comment]
> where would you prefer the previews to be? in the agep repo pr or here?

Both

**Comment by theartofnowt** [comment]
grand

File: README.md [hunk 1]
```diff
@@ -6,16 +6,20 @@ This package is distributed through [npm](https://www.npmjs.com/).
...
-Files in the base directory contain CSS which isn't targeted directly using a class but is compiled into the final CSS output (eg box-sizing or reset). It also contains usful mixins such as breakpoints.
+Files in the config directory contain CSS which isn't targeted directly using a class but is compiled into the final CSS output (eg box-sizing or reset). It also contains usful mixins such as breakpoints.
 
 ### Utitilities
 
 Files in utilities contains useful, compact snippets of code which can be accessed directly using a class in your HTML
 
+### Variables
+
+Files in variables contain specific variables for breakpoints and fonts.
```
Comment (ethancroft) [inline README.md]: contains* 
for breakpoints, fonts, etc.*

File: config/_config.mixins.scss [hunk 2]
```diff
@@ -49,3 +49,30 @@ Ideally, all mixins should be declared here but in some cases, they've been decl
     -moz-appearance: $value;
     appearance: $value;
 }
+
+/* clearfix
```
Comment (ethancroft) [inline config/_config.mixins.scss]: Why did we need this done as part of this branch?

I'm thinking this be ported into another branch todo for the future.

File: utilities/_utilities.clears.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
+.u-clearfix {
```
Comment (ethancroft) [inline utilities/_utilities.clears.scss]: Why did we need this done as part of this branch?

I'm thinking this be ported into another branch todo for the future.

File: README.md (hunk 1 above)
Comment (theartofnowt) [reply README.md]: Done - sort grammar out.

File: config/_config.mixins.scss (hunk 2 above)
Comment (theartofnowt) [reply config/_config.mixins.scss]: ok have transferred this to another branch for now

File: utilities/_utilities.clears.scss (hunk 3 above)
Comment (theartofnowt) [reply utilities/_utilities.clears.scss]: ok have transferred this to another branch for now

File: utilities/_utilities.clears.scss [hunk 4]
```diff
@@ -1,3 +1,4 @@
+//TODO: Make these into Mixins
```
Comment (ethancroft) [inline utilities/_utilities.clears.scss]: `// TODO: Convert into mixins`

Just cleaning up formatting to match `images.scss` todo

Comment (theartofnowt) [reply utilities/_utilities.clears.scss]: done

File: README.md [hunk 5]
```diff
@@ -6,16 +6,20 @@ This package is distributed through [npm](https://www.npmjs.com/).
...
-There are currently two directories, base and utilities. They are both intended to be used on any project but the key difference is how the code is used.
+There are currently three directories, config, variables and utilities. They are intended to be used on any project but the key difference is how the code is used.
 
-### Base
+### Config
 
-Files in the base directory contain CSS which isn't targeted directly using a class but is compiled into the final CSS output (eg box-sizing or reset). It also contains usful mixins such as breakpoints.
+Files in the config directory contain CSS which isn't targeted directly using a class but is compiled into the final CSS output (eg box-sizing or reset). It also contains usful mixins such as breakpoints.
 
 ### Utitilities
```
Comment (ethancroft) [inline README.md]: Weird spelling of utilities (I know you didn't do this but one to fix!)

Can you put this section at below Variables and add a note saying these file must be imported below the above partials.

File: README.md [hunk 6]
```diff
@@ -6,16 +6,22 @@ This package is distributed through [npm](https://www.npmjs.com/).
...
-### Utitilities
+### Config
+
+Files in the config directory contain CSS which isn't targeted directly using a class but is compiled into the final CSS output (eg box-sizing or reset). It also contains usful mixins such as breakpoints.
+
+### Utilities
 
 Files in utilities contains useful, compact snippets of code which can be accessed directly using a class in your HTML
 
+*Note* These files must be one of the last imported partials
```
Comment (ethancroft) [inline README.md]: Individual utility files or the main `utilities.scss` must be one of the last imported partials

Comment (theartofnowt) [reply README.md]: ok done

File: _all.scss [hunk 7]
```diff
@@ -1,18 +1,8 @@
-@import "base/mixins";
-@import "base/reset";
-@import "base/breakpoints";
-@import "base/box-sizing";
-@import "base/images";
-@import "base/flexbox";
+//variables
```
[the remark this answers is not in the document]
Comment (beccadawson) [reply _all.scss]: Happy with this

File: variables/_variables.breakpoints.scss [hunk 8]
```diff
@@ -0,0 +1,9 @@
+  // These variables are the default AP variables and can be overridden in other brand kits
```
Comment (beccadawson) [inline variables/_variables.breakpoints.scss]: Plz can you move this dev note into the Variables section of the readme?

File: README.md [hunk 9]
```diff
@@ -6,16 +6,22 @@ This package is distributed through [npm](https://www.npmjs.com/).
...
+There are currently three directories, config, variables and utilities. They are intended to be used on any project but the key difference is how the code is used.
 
-### Base
+### Variables
 
-Files in the base directory contain CSS which isn't targeted directly using a class but is compiled into the final CSS output (eg box-sizing or reset). It also contains usful mixins such as breakpoints.
+Files in variables contains specific variables for breakpoints, fonts, etc.
 
-### Utitilities
+### Config
```
[the remark this answers is not in the document]
Comment (beccadawson) [reply README.md]: And maybe there should be a distinction between base and mixins?

File: _all.scss [hunk 10]
```diff
@@ -1,18 +1,18 @@
...
+This file imports all 3 main partials into one file. This is required for legacy projects still using all.scss.
+
+Using _all.scss (which historically is one of the first partials to be imported in a project's scss config file) is not recommended as it opens up the risk of classes in utilities/_utilities.scss being overwritten by other classes declared in the stylesheet with the same CSS specificity score.
+
+For future use, import the below partials individually, with the utilities partial imported close to bottom of your config stylesheet.
+
+*/
+
+// Variables
+@import "variables/variables";
```
Comment (ethancroft) [inline _all.scss]: Is this not just `@import "variables"`?

File: _all.scss [hunk 11]
```diff
@@ -1,18 +1,18 @@
...
+
+// Base
+@import "base/base";
```
Comment (ethancroft) [inline _all.scss]: Is this not just `@import "base"`?

File: _all.scss [hunk 12]
```diff
@@ -1,18 +1,18 @@
...
+
+// Utilities
+@import "utilities/utilities";
```
Comment (ethancroft) [inline _all.scss]: Is this not just `@import "utilities"`?

**Comment by theartofnowt** [comment]
Ok, i have been on the repos that require `node_modules/oleg-starter-pack/all"` inputted this branch and did an `npm update` ran `gulp`. no confirmed issues - all repos update accordingly some are reallllllll old.

File: config/_config.box-sizing.scss [hunk 13]
```diff
@@ -3,4 +3,3 @@
     -moz-box-sizing: border-box;    /* Firefox, other Gecko */
     box-sizing: border-box;         /* Opera/IE 8+ */
 }
```
[the remark this answers is not in the document]
Comment (theartofnowt) [reply config/_config.box-sizing.scss]: done

File: _all.scss (hunk 12 above)
Comment (theartofnowt) [reply _all.scss]: done

File: _all.scss (hunk 11 above)
Comment (theartofnowt) [reply _all.scss]: done

File: _all.scss (hunk 10 above)
Comment (theartofnowt) [reply _all.scss]: done

File: variables/_variables.breakpoints.scss (hunk 8 above)
Comment (theartofnowt) [reply variables/_variables.breakpoints.scss]: done

File: _all.scss [hunk 14]
```diff
@@ -1,18 +1,8 @@
...
-@import "utilities/padding";
-@import "utilities/position";
-@import "utilities/hidden-elements";
-@import "utilities/images";
-@import "utilities/text-align";
-@import "utilities/transforms";
-@import "utilities/line-break";
-@import "utilities/font-weight";
\ No newline at end of file
+//config
```
[the remark this answers is not in the document]
Comment (theartofnowt) [reply _all.scss]: done

File: README.md [hunk 15]
```diff
@@ -6,16 +6,22 @@ This package is distributed through [npm](https://www.npmjs.com/).
...
+
+Files in the config directory contain CSS which isn't targeted directly using a class but is compiled into the final CSS output (eg box-sizing or reset). It also contains usful mixins such as breakpoints.
+
+### Utilities
 
 Files in utilities contains useful, compact snippets of code which can be accessed directly using a class in your HTML
 
+*Note* Individual utility files or the main <code>utilities.scss</code> must be one of the last imported partials
```
[the remark this answers is not in the document]
Comment (theartofnowt) [reply README.md]: done

File: README.md [hunk 16]
```diff
@@ -6,16 +6,22 @@ This package is distributed through [npm](https://www.npmjs.com/).
 
 ## File structure
 
-There are currently two directories, base and utilities. They are both intended to be used on any project but the key difference is how the code is used.
+There are currently three directories, config, variables and utilities. They are intended to be used on any project but the key difference is how the code is used.
```
[the remark this answers is not in the document]
Comment (theartofnowt) [reply README.md]: done

File: README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (theartofnowt) [reply README.md]: done - but not sure regarding the base and mixins

**Comment by ethancroft** [comment]
@marymooney Chiming in here!

STL buttons have always had a border-radius and are [live](https://www.agepartnership.co.uk/mcx/equity-release/do:landing/mc0:InboundReferral/mc1:StandardLife/mc2:CalculatorWidget/) atm - I've very recently refactored the partner brand kits so they all use the same `$button-border-radius` variable! 

Suspect your own testing branch has out of date npm packages and MBs doesn't!

**Comment by theartofnowt** [comment]
Gonna update the web.agep branch and rid merge conflictos.

**Comment by theartofnowt** [comment]
@beccadawson @marymooney @ethancroft updated https://github.com/AgePartnership/web.agepartnership.co.uk/pull/2373 merge conflicts sorted if need reviewed

**Pull request merged**